### ISSUE TITLE
Textbox Autoheight Logic

### ISSUE THREAD

**Issue by FarukOzderim** [gradio-app/gradio#980]
Continuation of halted discussion at #929 

I propose having `lines=1` by default, always having autoheight logic for `textbox` component to increase its size with the incoming input. What are your thoughts @abidlabs @aliabid94 @pngwn?

```
@abidlabs @pngwn I think having input textbox lines dynamically changing would be great as well. It would make it much more less cumbersome, WDYT?

Guess it depends how picky users are about how the component is laid out. We could use the autoheight logic when a lines value isn't provided?

Yeah I thought the same thing at first, then later thought what about respecting the line size initially and let the line size increase when input does not fit in.
Btw the default lines value is 1 currently, but we could change it.
```

**Comment by abidlabs** [comment]
I don't have a strong opinion on this. I can see how increasing the input textbox height can be useful in some cases, but in other cases (imagine a large text that is copy-pasted), then it could lead to the textbox becoming very large and crowding out the other components.

**Comment by FarukOzderim** [comment]
> I don't have a strong opinion on this. I can see how increasing the input textbox height can be useful in some cases, but in other cases (imagine a large text that is copy-pasted), then it could lead to the textbox becoming very large and crowding out the other components.

Is large text a use case though?
If it is then we shan't enable it by default or make it closable by parameters. 
Then how does this sound?

- If lines is not provided we make it autoheight
- If autoheight is True, we make it autoheight
- otherwise autoHeight is False

**Comment by pngwn** [comment]
We could also set a max pretty easily.

**Comment by abidlabs** [comment]
I assume that should be `autoheight=False` above. 

Otherwise, seems reasonable. 

I think we can wait until after 3.0 as this doesn't seem be urgent.

**Comment by FarukOzderim** [comment]
Would like to include in 3.0 if it is a short work from frontend perspective. Otherwise can wait in low priority.

**Comment by pngwn** [comment]
Frontend logic is already there really (for outputs), just needs hooking up to whatever param we want to use to control it. I think there should be a default max_height/ max_lines otherwise weird stuff might happen. Probably around 10.

**Comment by FarukOzderim** [comment]
Ok then, I will update accordingly.

**Comment by abidlabs** [comment]
Sorry one quick thing:

If we have a `lines` parameter and a `max_lines` parameter, do we need `autoheight`? We can just allow a user to block auto-adjusting by setting `max_lines` equal to `lines`? We can cutdown on one unnecessary parameter and prevent parameter settings from clashing

**Comment by FarukOzderim** [comment]
Agree with @abidlabs, lets use `lines` as max_lines.

autoheight is enables by default:

- when autoheight is True, min_lines is 1 and max_lines is `lines`
- when autoheight is False, min_lines = max_lines = `lines`

How does this sound @abidlabs and @pngwn

**Comment by FarukOzderim** [comment]
Moving this to 3.0 milestone since it is implemented in the backend #1009, feel free to move it to 2023 milestone.
